src/shared/buttercup/entries.js

```diff
@@ -57,6 +57,18 @@ export function validateEntry(entry) {
     if (fields.filter(field => !field.property).length > 0) {
       errorMessages.push(i18n.t('entry.custom-fields-label-empty-info'));
     }
+
+    if (fields.length > 3) {
```

Comment:
number of "default fields" are not always 3. 
You can find the custom fields like so:

```js
const removeableFields = fields.filter(field => field.removeable);
if (removeableFields.length > 0) {
```